systemservices/ethwallet/client/encoding.go

```diff
@@ -1,13 +1,27 @@
 package client
 
-import "encoding/json"
+import (
+	"encoding/json"
+
+	"github.com/mitchellh/mapstructure"
+)
 
 // Marshal returns the MESG format encoding of v.
-func Marshal(v interface{}) ([]byte, error) {
-	return json.Marshal(v)
+func Marshal(input interface{}) (map[string]interface{}, error) {
+	b, err := json.Marshal(input)
+	if err != nil {
+		return nil, err
+	}
+
+	var output map[string]interface{}
+	if err := json.Unmarshal(b, &output); err != nil {
+		return nil, err
+	}
+
+	return output, nil
```

Comment:
why marshal and then unmarshal?

Reply:
Just to convert `interface{}` to `map[stirng]interface{}`.

It's not the best client out there, but api is changin and now when it's kind of stable we can think of client api